tests/tape/tapes/test_tape.py

```diff
@@ -778,6 +781,65 @@ def test_measurement_expansion(self):
         expected = [None, [1, -1, -1, 1], [0, 5]]
         assert [m.eigvals is r for m, r in zip(new_tape.measurements, expected)]
 
+    def test_expand_tape_multiple_wires(self):
+        """Test the expand() method when measurements with more than one observable on the same
+        wire are used"""
+        with QuantumTape() as tape1:
+            qml.RX(0.3, wires=0)
+            qml.RY(0.4, wires=1)
+            qml.expval(qml.PauliX(0))
+            qml.var(qml.PauliX(0) @ qml.PauliZ(1))
+
+        with QuantumTape() as tape2:
+            qml.RX(0.3, wires=0)
+            qml.RY(0.4, wires=1)
+            qml.RY(-np.pi / 2, wires=0)
+            qml.expval(qml.PauliZ(0))
+            qml.var(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        tape1_exp = tape1.expand(expand_measurements=True)
+        tape2_exp = tape2.expand(expand_measurements=True)
+
+        assert tape1_exp.graph.hash == tape2_exp.graph.hash
+
+    def test_expand_tape_multiple_wires_non_commuting(self):
+        """Test if a QuantumFunctionError is raised during tape expansion if expectation values of
+        non-commuting observables are on the same wire"""
+        with QuantumTape() as tape:
+            qml.RX(0.3, wires=0)
+            qml.RY(0.4, wires=1)
+            qml.expval(qml.PauliX(0))
+            qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(qml.QuantumFunctionError, match="Only observables that are qubit-wise"):
+            tape.expand(expand_measurements=True)
+
+    def test_expand_tape_multiple_wires_non_commuting_var(self):
+        """Test if a QuantumFunctionError is raised during tape expansion if variances and
+        expectation values of non-commuting observables are on the same wire"""
+
+        with QuantumTape() as tape:
+            qml.RX(0.3, wires=0)
+            qml.RY(0.4, wires=1)
+            qml.expval(qml.PauliX(0))
+            qml.var(qml.PauliZ(0))
+
+        with pytest.raises(qml.QuantumFunctionError, match="Only observables that are qubit-wise"):
+            tape.expand(expand_measurements=True)
+
+    def test_expand_tape_multiple_wires_non_commuting_sample(self):
+        """Test if a QuantumFunctionError is raised during tape expansion if samples and
+        expectation values of non-commuting observables are evaluated on the same wire"""
+
+        with QuantumTape() as tape:
+            qml.RX(0.3, wires=0)
+            qml.RY(0.4, wires=1)
+            qml.expval(qml.PauliX(0))
+            qml.sample(qml.PauliZ(0))
```

Comment:
This is a case where I probably _would_ parametrize the tests with the measurement function, since it doesn't change any of the internal logic

Reply:
Thanks, done!